### PR TITLE
Hotfix build

### PR DESCRIPTION
**Pull request by xolian**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -8,10 +8,20 @@
 
 
 PIP = "pip"
+PYINSTALLER = "pyinstaller"
 BASE_DIR = os.path.dirname(os.path.abspath(__file__))
 REQUIREMENTS_FILE = os.path.join(os.path.abspath(BASE_DIR), 'requirements.txt')
 SETUP_FILE = os.path.join(os.path.abspath(BASE_DIR), 'setup.py')
 WEBBREAKER_MAIN = os.path.join(os.path.abspath(BASE_DIR), 'webbreaker', '__main__.py')
+DMG_FILE= os.path.join(os.path.abspath(BASE_DIR), 'dist', 'webbreaker')
+# distro = sys.platform
```
Comment (KylerWitting) [inline build.py]: Is comment necessary?

File: build.py [hunk 2]
```diff
@@ -8,10 +8,20 @@
...
+distro = platform.dist()[0]
+#distro_major_version = platform.dist()[1].split('.')[0]
```
Comment (KylerWitting) [inline build.py]: Remove comment block

File: build.py [hunk 3]
```diff
@@ -21,12 +31,13 @@
     SITEPACKAGES_RES = Popen(SITEPACKAGES_CMD, stdout=PIPE)
     SITEPACKAGES = str(SITEPACKAGES_RES.communicate()[0].decode()).rstrip()
 
-    PYINSTALLER_CMD = ['pyinstaller', "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--name", "webbreaker",
-                       "-p",
-                       SITEPACKAGES, WEBBREAKER_MAIN]
+    PYINSTALLER_CMD = [PYINSTALLER, "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--name", "webbreaker",
+                       "-p", SITEPACKAGES, WEBBREAKER_MAIN]
     # Make a macos dmg file
+    DMG_FILE = os.path.join(os.path.abspath(BASE_DIR), 'dist', 'webbreaker')
```
Comment (KylerWitting) [inline build.py]: Same path is set by default on line 16

File: build.py [hunk 4]
```diff
@@ -35,18 +46,18 @@
...
-    #
-    RPMBUILD_DIR = os.path.join(os.path.abspath(BASE_DIR), 'rpmbuild', 'SOURCES', 'webbreaker-2.0', 'opt', 'webbreaker')
-    PYINSTALLER_CMD = ['pyinstaller', "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--dist", RPMBUILD_DIR,
-                       "--name", "webbreaker-cli", "-p",
+    # Destination to put executable for bdist_rpm
+    RPMBUILD_DIR = os.path.join(os.path.abspath(BASE_DIR), 'dist')
+    PYINSTALLER_CMD = [PYINSTALLER, "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--dist", RPMBUILD_DIR,
+                       "--name", "webbreaker.linux", "-p",
                        SITEPACKAGES, WEBBREAKER_MAIN]
+    #sys.path.append(os.path.expanduser('~/.local/bin'))
```
Comment (KylerWitting) [inline build.py]: Remove comment unless necessary

File: build.py [hunk 5]
```diff
@@ -72,15 +82,25 @@ def cmdline(command):
...
                     # Run install requirements.txt for pyinstaller
                     if os.path.isfile(REQUIREMENTS_FILE):
-                        check_output(REQ_INSTALL, shell=True)
+                        print(cmdline(REQ_INSTALL))
                         # Install and run pyinstaller
-                        check_output(PYINSTALLER_REQ, shell=True)
-                        check_output(PYINSTALLER_CMD)
-                        cmdline(HDIUTIL_CMD)
+                        print("Your distro is: ---> {0} version {1} <---".format(sys.platform, os.uname()[2]))
+                        #print(check_output(["pip", "install", "--user", "PyInstaller==3.3"], shell=True))
```
Comment (KylerWitting) [inline build.py]: Remove comments

File: build.py [hunk 6]
```diff
@@ -72,15 +82,25 @@ def cmdline(command):
...
+                        try:
+                            print(cmdline(PYINSTALLER_REQ))
+                            print(cmdline(PYINSTALLER_CMD))
+                        except OSError:
+                            print(
+                                "There was an issue locating pyinstaller, you can run this command manually --> {}".format(
```
Comment (KylerWitting) [inline build.py]: Clean up printing of command

File: build.py [hunk 7]
```diff
@@ -35,18 +42,18 @@
...
-    #
-    RPMBUILD_DIR = os.path.join(os.path.abspath(BASE_DIR), 'rpmbuild', 'SOURCES', 'webbreaker-2.0', 'opt', 'webbreaker')
-    PYINSTALLER_CMD = ['pyinstaller', "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--dist", RPMBUILD_DIR,
-                       "--name", "webbreaker-cli", "-p",
+    # Destination to put executable for bdist_rpm
+    RPMBUILD_DIR = os.path.join(os.path.abspath(BASE_DIR), 'dist')
+    PYINSTALLER_CMD = [PYINSTALLER, "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--dist", RPMBUILD_DIR,
+                       "--name", "webbreaker.linux", "-p",
                        SITEPACKAGES, WEBBREAKER_MAIN]
+    #sys.path.append(os.path.expanduser('~/.local/bin'))
```
Comment (KylerWitting) [inline build.py]: Remove commented out code

File: build.py [hunk 8]
```diff
@@ -72,15 +78,25 @@ def cmdline(command):
...
                     # Run install requirements.txt for pyinstaller
                     if os.path.isfile(REQUIREMENTS_FILE):
-                        check_output(REQ_INSTALL, shell=True)
+                        cmdline(REQ_INSTALL)
                         # Install and run pyinstaller
-                        check_output(PYINSTALLER_REQ, shell=True)
-                        check_output(PYINSTALLER_CMD)
-                        cmdline(HDIUTIL_CMD)
+                        print("Your distro is: ---> {0} version {1} <---".format(sys.platform, os.uname()[2]))
+                        #print(check_output(["pip", "install", "--user", "PyInstaller==3.3"], shell=True))
```
Comment (KylerWitting) [inline build.py]: Remove commented out code

File: build.py [hunk 9]
```diff
@@ -21,12 +27,13 @@
     SITEPACKAGES_RES = Popen(SITEPACKAGES_CMD, stdout=PIPE)
     SITEPACKAGES = str(SITEPACKAGES_RES.communicate()[0].decode()).rstrip()
 
-    PYINSTALLER_CMD = ['pyinstaller', "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--name", "webbreaker",
-                       "-p",
-                       SITEPACKAGES, WEBBREAKER_MAIN]
+    PYINSTALLER_CMD = [PYINSTALLER, "--clean", "-y", "--windowed", "--noconsole", "--onefile", "--name", "webbreaker",
+                       "-p", SITEPACKAGES, WEBBREAKER_MAIN]
     # Make a macos dmg file
+    DMG_FILE = os.path.join(os.path.abspath(BASE_DIR), 'dist', 'webbreaker')
```
Comment (KylerWitting) [inline build.py]: Same path is set by default on line 16

File: build.py [hunk 10]
```diff
@@ -8,10 +8,16 @@
 
 
 PIP = "pip"
+PYINSTALLER = "pyinstaller"
 BASE_DIR = os.path.dirname(os.path.abspath(__file__))
 REQUIREMENTS_FILE = os.path.join(os.path.abspath(BASE_DIR), 'requirements.txt')
 SETUP_FILE = os.path.join(os.path.abspath(BASE_DIR), 'setup.py')
 WEBBREAKER_MAIN = os.path.join(os.path.abspath(BASE_DIR), 'webbreaker', '__main__.py')
+DMG_FILE= os.path.join(os.path.abspath(BASE_DIR), 'dist', 'webbreaker')
+# distro = sys.platform
```
Comment (KylerWitting) [inline build.py]: Remove commented out code

Comment (xolian) [reply build.py]: done

File: build.py (hunk 9 above)
Comment (xolian) [reply build.py]: done

File: build.py (hunk 7 above)
Comment (xolian) [reply build.py]: done

File: build.py [hunk 11]
```diff
@@ -2,90 +2,89 @@
...
+                    except OSError:
+                        print(
+                            "There was an issue locating pyinstaller, you can run this command manually --> "
+                            "{0}{1}{2}".format("pyinstaller --clean -y --windowed --onefile --name webbreaker -p ",
+                                               SITEPACKAGES, WEBBREAKER_MAIN))
+
+                    if distro == "darwin":
+                        try:
+                            cmdline(HDIUTIL_CMD)
+                            print("Successfully built the mac installer {}.dmg!".format(DMG_FILE))
```
Comment (KylerWitting) [inline build.py]: Success message is printing when failure was reported also

```➜ python setup.py pyinstaller            
Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
There was an issue locating pyinstaller, you can run this command manually --> pyinstaller --clean -y --windowed --onefile --name webbreaker -p /Users/z002v51/.local/lib/python2.7/site-packages/Users/z002v51/Documents/webbreaker/webbreaker/__main__.py
hdiutil: create failed - No such file or directory
Successfully built the mac installer /Users/z002v51/Documents/webbreaker/dist/webbreaker.dmg!```

Comment (xolian) [reply build.py]: added...
                try:
                    # Install openssl, wheel and pyinstaller
                    check_output(open_ssl_module, shell=True)
                    check_output(wheel_module, shell=True)
                    # Run requirements
                    if os.path.isfile(REQUIREMENTS_FILE):
                        ...
                    else:
                        sys.stderr.write("{} does not exist\n".format(REQUIREMENTS_FILE))
                        exit(1)
                except OSError:
                    print(
                        "There was an issue installing the python requirements and executing pyinstaller, you can run "
                        "these commands manually --> \npip install --user -r{0}\n{1}{2}{3}".format(
                            "pyinstaller --clean -y --windowed --onefile --name webbreaker -p ",
                            REQUIREMENTS_FILE, sitepackages, WEBBREAKER_MAIN, REQUIREMENTS_FILE))
                    exit(1)

**Pull request merged**